headers_test.go

```diff
@@ -614,6 +646,194 @@ func TestUnprotectedHeader_MarshalCBOR(t *testing.T) {
 			},
 			wantErr: "unprotected header: header parameter: crit: not allowed",
 		},
+		{
+			name: "malformed counter signature",
+			h: UnprotectedHeader{
+				HeaderLabelCounterSignature: "",
+			},
+			wantErr: "unprotected header: header parameter: counter signature is not a Countersignature or a list",
+		},
+		{
+			name: "counter signature without signature",
+			h: UnprotectedHeader{
+				HeaderLabelCounterSignature: []Countersignature{
+					{
+						Headers: Headers{
+							Protected: ProtectedHeader{
+								HeaderLabelAlgorithm: AlgorithmEd25519,
+							},
+							Unprotected: UnprotectedHeader{
+								HeaderLabelKeyID: []byte("11"),
+							},
+						},
+					},
+				},
+			},
+			wantErr: "empty signature",
+		},
+		{
+			name: "complete counter signature",
+			h: UnprotectedHeader{
+				HeaderLabelCounterSignature: []Countersignature{
+					{
+						Headers: Headers{
+							Protected: ProtectedHeader{
+								HeaderLabelAlgorithm: AlgorithmEd25519,
+							},
+							Unprotected: UnprotectedHeader{
+								HeaderLabelKeyID: []byte("11"),
+							},
+						},
+						Signature: []byte{
+							0xb7, 0xca, 0xcb, 0xa2, 0x85, 0xc4, 0xcd, 0x3e,
+							0xd2, 0xf0, 0x14, 0x6f, 0x41, 0x98, 0x86, 0x14,
```

Comment:
potentially blocks like this, can be imported directly from files like https://github.com/cose-wg/Examples/blob/3221310e2cf50ad13213daa7ca278209a8bc85fd/sign1-tests/sign-pass-02.json

Reply:
I understand your point, but these examples are outdated, as said, and they include changes to map the mocked signature algorithm.